### PR TITLE
mvcc: add metrics to characterize backend expensive reads

### PR DESCRIPTION
**Pull request by jingyih**
Metrics added:

etcd_debugging_mvcc_put_total
etcd_debugging_mvcc_put_duration_seconds
etcd_debugging_mvcc_range_total
etcd_debugging_mvcc_range_duration_seconds
etcd_debugging_mvcc_committed_read_total
etcd_debugging_mvcc_committed_read_duration_seconds
etcd_debugging_backend_batch_interval_duration_seconds

### CODE REVIEW

File: mvcc/backend/backend.go [hunk 1]
```diff
@@ -315,6 +316,8 @@ func (b *backend) run() {
 		if b.batchTx.safePending() != 0 {
 			b.batchTx.Commit()
 		}
+		batchIntervalSec.Observe(time.Since(start).Seconds() * 1e3)
```
Comment (jpbetz) [inline mvcc/backend/backend.go]: I believe `.Seconds()` already performs the needed unit conversion and we don't need the `* 1e3`.

File: mvcc/backend/metrics.go [hunk 2]
```diff
@@ -83,6 +83,17 @@ var (
 		// highest bucket start of 0.01 sec * 2^16 == 655.36 sec
 		Buckets: prometheus.ExponentialBuckets(.01, 2, 17),
 	})
+
+	batchIntervalSec = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Namespace: "etcd_debugging",
+		Subsystem: "backend",
+		Name:      "batch_interval_duration_milliseconds",
```
Comment (jpbetz) [inline mvcc/backend/metrics.go]: variable name suggests seconds but metric name suggests milliseconds.  For prometheus it's okay to use seconds here. There's a convention: https://prometheus.io/docs/practices/naming/#base-units

File: mvcc/backend/metrics.go [hunk 3]
```diff
@@ -83,6 +83,17 @@ var (
...
+		Help:      "The distributions of batch interval in mvcc backend.",
+
+		// lowest bucket start of upper bound 0.1 msec (10 us) with factor 2
+		// highest bucket start of 0.1 msec * 2^13 == 819.2 msec
+		Buckets: prometheus.ExponentialBuckets(0.1, 2, 14),
```
Comment (jpbetz) [inline mvcc/backend/metrics.go]: If we switch to seconds, we'll need to update the buckets to match.

**Review by jpbetz — CHANGES_REQUESTED** [review]

File: mvcc/backend/backend.go (hunk 1 above)
Comment (jingyih) [reply mvcc/backend/backend.go]: Switched to seconds.

**Pull request merged**